src/main/java/entities/hazards/ChasingEnemy.java

```diff
@@ -0,0 +1,79 @@
+package entities.hazards;
+
+import use_cases.hazards.IEnemyRequestModel;
```

Comment:
Same problem as before, there is still a clean architecture violation. Inner layers should know nothing from the outer layers, which means that entities should not be accessing any other layers since it is the innermost one. I know it's pretty annoying especially since we are working with a game, but we should do our best to follow the clean architecture structure.

Reply:
I think this is fine: there is no real dependence here, since IEnemyRequestModel is an interface. In the clean architecture, inner layers are supposed to access outer layers through interfaces (with dependency inversion).

Reply:
If you want, I can move the interface to the entities folder.

Reply:
I agree - by making it an interface, Chasing enemy doesn't depend on Player. It instead depends on the interface IEnemyRequestModel. Classic dependency inversion.

For the sake of organization though, maybe it's a good idea to move the interface to the entities folder like you guys said.

Reply:
In the case of dependency inversion I think it's best you have it in the same layer as the inner one, just like you mentioned. I was a little bit confused because request models are usually in the use cases layer, but I don't know too much about your implementation details, if you decided to put the request model in the entities layer and it works fine, there should be no problem.